teachers_digital_platform/tests/models/test_activity_index_page.py

```diff
@@ -69,95 +80,121 @@ def test_client_can_create_default_home_page(self):
-    #         student_characteristics=[],
-    #         activity_type=ActivityType.objects.filter(pk__in=[1, 2, 3]).all(),
-    #         teaching_strategy=ActivityTeachingStrategy.objects.filter(pk__in=[6, 7]).all(),
-    #         blooms_taxonomy_level=ActivityBloomsTaxonomyLevel.objects.filter(pk__in=[6]).all(),
-    #         activity_duration=ActivityDuration.objects.get(pk=2),
-    #         council_for_economic_education=ActivityCouncilForEconEd.objects.filter(pk__in=[4]).all(),
-    #         jump_start_coalition=ActivityJumpStartCoalition.objects.filter(pk__in=[1]).all()
-    #     )
-    #     self.search_page.add_child(instance=activity_page)
-    #     activity_page.save_revision().publish()
-    
-    
-    #     response = self.client.get(request.get_full_path() + "?q=voldemort")
-    #     self.assertContains(response, "<li>No results found.</li>", html=True)
\ No newline at end of file
+    @mock.patch('teachers_digital_platform.models.pages.SearchQuerySet.models')
+    def test_search_get_all_facets_with_building_block_filter(self, mock_sqs):
+        # Arrange
+        facet_counts = {
+            'dates': {},
+            'fields': {
+                    'topic':'1',
+                    'topic':'4',
+                    'building_block': '1',
+                    'building_block': '2',
+                    'school_subject': '1',
+                },
+                'queries': {}
+            }
+        selected_facets = {u'building_block': [1]}
+        facet_queries = {'building_block':'building_block_exact'}
+        facet_map = self.create_facet_map()
+        mock_sqs.facet_counts.return_value = facet_counts
+        # Act
+        actual_all_facets = self.search_page.get_all_facets(facet_map, mock_sqs, facet_counts, facet_queries, selected_facets)
+        # Assert
+        self.assertTrue('building_block' in actual_all_facets)
+
+    def create_facet_map(self):
+        return (
+            ('building_block', (ActivityBuildingBlock, False, 10)),
+            ('school_subject', (ActivitySchoolSubject, False, 25)),
+            ('topic', (ActivityTopic, True, 25)),
+            ('grade_level', (ActivityGradeLevel, False, 10)),
+            ('age_range', (ActivityAgeRange, False, 10)),
+            ('student_characteristics', (ActivityStudentCharacteristics, False, 10)),
+            ('activity_type', (ActivityType, False, 10)),
+            ('teaching_strategy', (ActivityTeachingStrategy, False, 25)),
+            ('blooms_taxonomy_level', (ActivityBloomsTaxonomyLevel, False, 25)),
+            ('activity_duration', (ActivityDuration, False, 10)),
+            ('jump_start_coalition', (ActivityJumpStartCoalition, False, 25)),
+            ('council_for_economic_education', (ActivityCouncilForEconEd, False, 25)),
+        )
+
+    def create_activity_detail_page(self, title='title', slug='slug'):
+        return ActivityPage(
+            live=True, 
+            title='Planning for future savings',
```

Comment:
@atuggle should this be title=title and not title='Planning for future savings'? What's the purpose of the title and slug parameters in the method if they aren't being used?